pkg/build/controller/strategy/util.go

```diff
@@ -705,6 +715,13 @@ func setupBuildVolumes(pod *corev1.Pod, buildVolumes []buildv1.BuildVolume, csiV
 		case buildv1.BuildVolumeSourceTypeConfigMap:
 			volumeSource.ConfigMap = buildVolume.Source.ConfigMap
 			mountConfigMapVolume(pod, &pod.Spec.Containers[0], strings.ToLower(buildVolume.Source.ConfigMap.Name), PathForBuildVolume(buildVolume.Source.ConfigMap.Name), buildVolumeSuffix, &volumeSource)
+		case buildv1.BuildVolumeSourceTypeCSI:
+			if !csiVolumesEnabled {
+				return fmt.Errorf("csi volume request cannot be fullfilled without enabling csivolume feature on the cluster")
```

Comment:
I would reference the feature gate name here. Admins can enable tech preview on the cluster to do this, OR provide the feature gate name with their own custom feature set:

```suggestion
				return fmt.Errorf("csi volumes require the BuildCSIVolumes feature gate to be enabled")
```